src/components/TitleBar.js

```diff
@@ -10,13 +10,15 @@ const TitleBar = ({ onPrevious, title, onDismiss, titleExtras }) => {
       display: 'flex', alignItems: 'baseline', flex: 'none', padding: '1.5rem'
     }
   }, [
-    onPrevious && h(Clickable, {
+    (onPrevious || onDismiss) && !(onPrevious && onDismiss) ? div({ style: { fontSize: '0.875rem', fontWeight: 600 } }, [title]) : undefined,
+    onPrevious && h(Link, {
       'aria-label': 'Back',
+      style: !onDismiss ? { marginLeft: 'auto' } : undefined,
       onClick: onPrevious
-    }, [icon('arrowLeft', { size: 17 })]),
-    div({ style: { fontSize: '0.875rem', fontWeight: 600, marginLeft: onPrevious ? 'auto' : undefined } }, [title]),
+    }, [icon('arrowLeft', { size: 20 })]),
+    (onPrevious && onDismiss) ? div({ style: { fontSize: '0.875rem', fontWeight: 600, marginLeft: onPrevious ? 'auto' : undefined } }, [title]) : undefined,
```

Comment:
This means the buttons will swap sides depending on the situation? That seems like it could be a confusing UX. What was the goal here?

Reply:
The goal was that if there was only going to be only one action available (either an 'x' exit or '<-' back) that this button would always be on the right side. This came from the updated mock up and a discussion with Lou, with the goal of a more consistent look to the UI

Reply:
mock up with updated look: https://projects.invisionapp.com/d/main?origin=v7#/console/16427254/383729206/preview

Reply:
Flipped it back after consultation with @zarsky-broad